gcsio/src/test/java/com/google/cloud/hadoop/gcsio/WatchdogTest.java

```diff
@@ -0,0 +1,342 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.lang.Thread.sleep;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static org.junit.Assert.assertThrows;
+
+import com.google.common.collect.Lists;
+import com.google.common.flogger.GoogleLogger;
+import com.google.storage.v2.ReadObjectResponse;
+import com.google.storage.v2.WriteObjectRequest;
+import com.google.storage.v2.WriteObjectResponse;
+import io.grpc.ClientCall;
+import io.grpc.Context;
+import io.grpc.Context.CancellableContext;
+import io.grpc.internal.NoopClientCall;
+import io.grpc.stub.StreamObserver;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeoutException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class WatchdogTest {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private Watchdog watchdog;
+
+  private final Duration waitTime = Duration.ofSeconds(5);
+  private final Duration zeroWaitTime = Duration.ofSeconds(0);
+
+  @Before
+  public void setUp() throws Exception {
+    Duration checkInterval = Duration.ofSeconds(2);
+    ScheduledExecutorService executor = newSingleThreadScheduledExecutor();
+    watchdog = Watchdog.create(checkInterval, executor);
+  }
+
+  @Test
+  public void testPassThroughClientStreamingRPC() {
```

Comment:
nit: do not use prefix `test` for test names - it's obsolete requirement from JUnit 3 era.

Reply:
Done